scalableminds/webknossos
# Folder And Dataset Metadata

### URL of deployed dev instance (used for testing):
- https://folderdatasetproperties.webknossos.xyz/

### Steps to test:
- Open the dashboard & create another folder.
- Try adding metadata to the folder. Try out all three kinds of entries. Test whether autosave works
- Try editing metadata for a dataset
- Switch between datasets, the metadata should stay consistent
- Switch between folders, the metadata should stay consistent
- And also switch between folders and datasets and also look for any errors / inconsistencies / lost metadata entries.

### TODOs:
- [x] Clean up git history 
- [x] Allow to search for metadata props -> Other issue https://github.com/scalableminds/webknossos/issues/8034
- [x] Add props & UI and everything for folders 
- [x] Update UI to the newest design proposal
- [x] Test dataset update route without supplying the metadata field 
  - [x] Works. Tested with firefox's edit and resend functionality. 

### Open Question
- [x] What about supporting dataset tags? Should they be removed? Should one still be able to update the dataset tags via the API? Should they be removed from the schema? Should they be transferred into metadata entries?
- [x] How should tags be transferred into metadata?
  - As a single string array entry with key "tags"
  - For each on entry with numbered keys like "tagX" and x is the index of the tag in the tag array
  - other ideas 


### Issues:
- fixes [#5968](https://github.com/scalableminds/webknossos/issues/5968)

------
(Please delete unneeded items, merge only when none are left open)
